src/import/import_mode.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use kvproto::import_sstpb::*;
+use rocksdb::DB;
+
+use super::Result;
+
+pub struct ImportModeSwitcher {
+    mode: SwitchMode,
+    backup_options: Vec<(String, ImportModeOptions)>,
+}
+
+impl ImportModeSwitcher {
+    pub fn new() -> ImportModeSwitcher {
+        ImportModeSwitcher {
+            mode: SwitchMode::Normal,
+            backup_options: Vec::new(),
+        }
+    }
+
+    pub fn enter_normal_mode(&mut self, db: &DB) -> Result<()> {
+        if self.mode == SwitchMode::Normal {
+            return Ok(());
+        }
+
+        for (cf_name, cf_opts) in &self.backup_options {
+            cf_opts.set_options_cf(db, cf_name)?;
+        }
+
```

Comment:
Should cleanup `backup_options` after have switched to normal mode?

Reply:
Not necessary, we will clear it before we backup again.